ant/src/balances.lua

```diff
@@ -0,0 +1,64 @@
+local utils = require("ant.src.utils")
+local json = require("ant.src.json")
+
+Balances = Balances or {}
+
+local balances = {}
+
+function balances.info(msg)
+	utils.reply(json.encode({
+		Name = Name,
+		Ticker = Ticker,
+		TotalSupply = 1,
+		Logo = Logo,
+		Denomination = tostring(Denomination),
+	}))
+end
+
+function balances.transfer(msg)
+	local from = msg.From
+	local to = msg.Tags.Recipient
+
+	local ownerStatus, ownerResult = pcall(utils.validateOwner, from)
+	local recipientStatus, recipientResult = pcall(utils.validateArweaveId, to)
+
+	if not ownerStatus then
+		return utils.reply(ownerResult)
+	else
+		if not recipientStatus then
+			return utils.reply(recipientResult)
+		else
+			Balances[from] = nil
+			Balances[to] = 1
+		end
+	end
+end
+
+function balances.balance(msg)
+	local address = msg.Tags.Address or msg.From
+	local balance = Balances[address]
+	if balance == nil then
+		return utils.reply("0")
+	end
+	utils.reply("1")
+end
```

Comment:
Just a suggestion these functions should just return the values and not use `utils`. Move all that to main.lua. That allows for more controlled testing of the actual inputs/outputs of functions without worrying about stubbing `utils`.

```suggestion
function balances.balance(address)
        assert(utils.validateArweaveId(address), "Addreess must be valid Arweave ID")
        local balance = Balances[address] or 0
        return 0
end
```